### DISCUSSION TITLE
Get filtermode localization in Column.Filter

### DISCUSSION THREAD

**Discussion by hippien** [KevinVandy/material-react-table#380] [General]
In the advanced example there is a custom Date Picker Filter from @mui/x-date-pickers on the "Start date" column. 
The table has 'enableColumnFilterModes'.

But the helperText in the columns Filter option is hard-coded to 'Filter Mode: Lesss Than' and will not update when the mode is change in the column filter menu.

(Try it here: [Advanced example](https://www.material-react-table.com/docs/examples/advanced#advanced-example))

Is it possible to access the filter mode localization text for the current column filter mode so it can be displayed correctly?

```
Filter: ({ column }) => (
              <LocalizationProvider dateAdapter={AdapterDayjs}>
                <DatePicker
                  onChange={(newValue) => {
                    column.setFilterValue(newValue);
                  }}
                  renderInput={(params) => (
                    <TextField
                      {...params}
                      helperText={'Filter Mode: Lesss Than'}
                      sx={{ minWidth: '120px' }}
                      variant="standard"
                    />
                  )}
                  value={column.getFilterValue()}
                />
              </LocalizationProvider>
            )
```

**Comment by KevinVandy** [comment]
There's something here: https://www.material-react-table.com/docs/guides/column-filtering#customize-filter-modes-source-code
that might help

<img width="696" alt="image" src="https://user-images.githubusercontent.com/28243511/219478277-e18b5a8e-efa4-4cba-80ef-d2925f4f76c7.png">

And I think there is some kind of bug there that needs to be looked into more

**Reply by hippien** [reply]
Not the cleanest solution but I imported the localization file:
```
import { MRT_Localization_EN } from 'material-react-table/locales/en';
const localization = MRT_Localization_EN;
```
Contructed the property name:

```
Filter: ({ column }) => {
          const s = column.columnDef.filterFn.name;
          const filterFnName = `filter${s ? s[0].toUpperCase() + s.slice(1) : ""}`;
```
And grabbed the text:
```
helperText={`Filter mode: ${localization[filterFnName]}`}
```
<img width="219" alt="image" src="https://user-images.githubusercontent.com/3788213/219494473-880d32e7-bd59-4cf7-b8ea-b4693d6ae68f.png">

A bit cumbersome and ugly but will do for now. Maybe there is a better way to it?

**Reply by KevinVandy** [reply]
no need to import `MRT_Localization_EN`. It already gets spread internally like this:
```js
_localization = { ...MRT_Localization_EN, ...props.localization }
```
